src/components/pdf-downloader/PDFDownloader.jsx

```diff
@@ -0,0 +1,32 @@
+import { saveAs } from 'file-saver'
+
+export const downloadPDF = async (pdfDocument, pdfUrl) => {
+  const pdfName = pdfUrl.substring(pdfUrl.lastIndexOf('/') + 1)
+
+  try {
+    // Try to download file via blob to avoid opening PDF in another PDF plugin
+    const pdfData = await pdfDocument.getData()
+    const blob = new Blob([pdfData], { type: 'application/pdf' })
+    saveAs(blob, pdfName)
+  } catch (error) {
+    // If blob is not supported or something happens
+    // download it via traditional way
+    // Code taken from https://github.com/mozilla/pdf.js/blob/master/web/download_manager.js#L28
+    const a = document.createElement('a')
+    a.href = pdfUrl
+    a.target = '_blank'
+
+    // Use a.download if available. This increases the likelihood that
+    // the file is downloaded instead of opened by another PDF plugin.
+    if ('download' in a) a.download = pdfName
+
+    // <a> must be in the document for IE and recent Firefox versions,
+    // otherwise .click() is ignored.
+    const rootDocument = document.body || document.documentElement
```

Comment:
The line logically seems to be incorrect
- `document.documentElement` is `<html>`
- `document.body` is actually `<body>`

Reply:
It's taken from [PDF.js](https://github.com/mozilla/pdf.js/blob/master/web/download_manager.js#L42). I don't see any reason for it either. I will remove it.

Reply:
It's up to you. I generally suggesting stuff, not requiring to change 🙂